server/middlewares/auth-sockets.js

```diff
@@ -0,0 +1,57 @@
+const { v4: uuid } = require('uuid');
+const { User } = require("../db/models");
+const cache = require("../services/cache");
+
+const auth = () => {
+  return async (socket, next) => {
+    try {
+      const sessionId = socket.handshake.auth.sessionId;
+
+      if (sessionId) {
+        const data = await cache.get(sessionId);
+        if (data && data.id) {
+          socket.sessionId = sessionId;
+          socket.userId = data.id;
+          return next();
+        }
+      }
+
+      const token = socket.handshake.auth.token;
+
+      if (!token) {
+        console.debug('Token not provided');
+        return next(new Error("Unathorized connection"));
+      }
+
+      const data = await cache.get(token);
+
+      if (data && data.id) {
+        const user = await User.findOne({
+          where: { id: data.id },
+        });
+
+        if (!user) {
+          console.debug('User not found');
+          return next(new Error("Unathorized connection"));
+        }
+
+        await cache.del(token);
+
+        const sessionId = uuid();
+        await cache.set(sessionId, "id", data.id);
+
+        socket.sessionId = sessionId;
+        socket.userId = user.id;
+        return next();
+      }
+
+      console.debug('Token not found');
```

Comment:
can remove all these console.logs in this file if you're done using them

Reply:
it was more a first step to transition to a robust logger like pino yet I can remove it